frame/wormhole/issuing/s2s/src/lib.rs

```diff
@@ -117,28 +117,23 @@ pub mod pallet {
 			let factory_id = <T as darwinia_evm::Config>::AddressMapping::into_account_id(factory);
 			ensure!(caller == factory_id, <Error<T>>::NotFactoryContract);
 
-			let burn_action = &sha3::Keccak256::digest(&BURN_ACTION)[0..4];
-			if &input[4..8] == burn_action {
-				let burn_info =
-					TokenBurnInfo::decode(&input[8..]).map_err(|_| Error::<T>::InvalidDecoding)?;
-				// Ensure the recipient is valid
-				ensure!(
-					burn_info.recipient.len() == 32,
-					<Error<T>>::InvalidAddressLen
-				);
-
-				let fee = Self::transform_dvm_balance(burn_info.fee);
-				if let Some(fee_account) = T::FeeAccount::get() {
-					// Since fee account will represent use to make a cross chain call, give fee to fee account here.
-					// the fee transfer path
-					// user -> mapping_token_factory(caller) -> fee_account -> fee_fund -> relayers
-					<T as Config>::RingCurrency::transfer(&caller, &fee_account, fee, KeepAlive)?;
-				}
+			let burn_info =
+				TokenBurnInfo::decode(&input[8..]).map_err(|_| Error::<T>::InvalidDecoding)?;
+			// Ensure the recipient is valid
+			ensure!(
+				burn_info.recipient.len() == 32,
+				<Error<T>>::InvalidAddressLen
+			);
 
-				Self::burn_and_remote_unlock(fee, burn_info)?;
-			} else {
-				log::trace!("No action match this input selector");
+			let fee = Self::transform_dvm_balance(burn_info.fee);
+			if let Some(fee_account) = T::FeeAccount::get() {
+				// Since fee account will represent use to make a cross chain call, give fee to fee account here.
+				// the fee transfer path
+				// user -> mapping_token_factory(caller) -> fee_account -> fee_fund -> relayers
+				<T as Config>::RingCurrency::transfer(&caller, &fee_account, fee, KeepAlive)?;
 			}
+
+			Self::burn_and_remote_unlock(fee, burn_info)?;
```

Comment:
https://github.com/darwinia-network/darwinia-common/issues/816

Need refactoring, may be in another PR.

Reply:
Leave this to another PR.